evalml/utils/woodwork_utils.py

```diff
@@ -85,29 +92,14 @@ def _retain_custom_types_and_initalize_woodwork(old_woodwork_data, new_pandas_da
         A new DataTable where any of the columns that exist in the old input DataTable and the new DataFrame try to retain
         the original logical type, if possible and not specified to be ignored.
     """
-    if isinstance(old_woodwork_data, ww.DataColumn):
-        if str(new_pandas_data.dtype) != old_woodwork_data.logical_type.pandas_dtype:
-            try:
-                return ww.DataColumn(new_pandas_data, logical_type=old_woodwork_data.logical_type)
-            except (ValueError, TypeError):
-                pass
-        return ww.DataColumn(new_pandas_data)
-
-    retained_logical_types = {}
+    if isinstance(new_dataframe, pd.Series):
+        return ww.init_series(new_dataframe, old_logical_types)
     if ltypes_to_ignore is None:
         ltypes_to_ignore = []
-    col_intersection = set(old_woodwork_data.columns).intersection(set(new_pandas_data.columns))
```

Comment:
Since we're no longer converting from nullable to non-nullable types, I think we can simplify the implementation. 

I tested this on my branch that updates the entire repo and I think this implementation meets our requirements.